src/preferences/device.js

```diff
@@ -306,15 +309,30 @@ var Panel = GObject.registerClass({
             path: `/org/gnome/shell/extensions/gsconnect/device/${device.id}/`,
         });
 
-        // Infobar
+        // Pairing infobar
         this.device.bind_property(
             'paired',
             this.infobar,
             'reveal-child',
             (GObject.BindingFlags.SYNC_CREATE |
              GObject.BindingFlags.INVERT_BOOLEAN)
         );
+        this.device.connect('notify::pairing', () => {
+            if (this.device.pairing)
+                this.infobar_stack.visible_child = this.infobar_verify;
+            else
+                this.infobar_stack.visible_child = this.infobar_pair;
+
+        });
+
+        this.device.bind_property(
+            'verify-code',
+            this.verifycode,
+            'label',
+            GObject.BindingFlags.SYNC_CREATE
+        );
 
+        // Action setup
         this._setupActions();
```

Comment:
Phew, there's that mystery solved! :smiling_imp:

Reply:
Lol, I recall feeling the same way writing it. But it looked out of place otherwise 😂